core/src/test/kotlin/generator/test/GeneratorTest.kt

```diff
@@ -0,0 +1,53 @@
+package generator.test
+
+import generator.ConfigParams
+import generator.TestRunner
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.junit.Test
+import java.nio.file.Path
+
+class GeneratorTest {
+
+    val logger = DokkaConsoleLogger
+    fun config(): (ConfigParams) -> DokkaConfiguration = { (name, root, out) ->
+        DokkaConfigurationImpl(
+            outputDir = out.toString(),
+            format = "html",
+            generateIndexPages = true,
+            passesConfigurations = listOf(
+                PassConfigurationImpl(
+                    moduleName = name,
+                    sourceRoots = listOf(root).map { SourceRootImpl(it.toString()) },
+                    analysisPlatform = Platform.jvm,
+                    targets = listOf("jvm")
+                )
+            )
+        )
+    }
+
+        @Test
+        fun test1() {
+            val checks: Map<DokkaTestGenerator.Companion.Stage, (DokkaTestGenerator.State) -> Unit?> = mapOf(
+                DokkaTestGenerator.Companion.Stage.DOCUMENTABLES_CREATION to { state: DokkaTestGenerator.State ->
+                    state.context?.platforms?.forEach { (pd, _) ->
+                        logger.info(pd.name)
+                    }
+                },
+                DokkaTestGenerator.Companion.Stage.RENDERING to { state: DokkaTestGenerator.State ->
+                    val test = state.pages?.parentMap?.size == 7
+                    if (!test) println(state.pages?.parentMap?.size)
+                    assert(test)
+                }
+            )
+
+            TestRunner.generateTestDokka(name = "test1", query = """
```

Comment:
`Name` should be extracted automatically

Reply:
Done.